Docs: Update Developer Guide (#309)
AY2324S2-CS2103T-T17-3/tp: docs/DeveloperGuide.md

![DeleteCommandActivityDiagram](images/DeleteCommandActivityDiagram.png)

### Additional user details in FitBook

> On top of what AB3 has to offer, FitBook allows users to add additional details to each client to better track their health status. Some key features include:

Reviewer: Should we mention AB3?

> The above sequence diagram also applies to the removal of a note from an existing client when no input string or prefix is entered for the `note` command. (i.e. `note 1`, or `note 1 nt/`).

#### Weight tracking feature

> The weight tracking feature allows users to keep track of past weight measurements of a client. Since a client is a person, we have set a more-than-reasonable limit of 5000kg for `weight`.

Reviewer 2: I think we should compile all the valid parameters together in a table and just reference that table when needed.

> The weight tracking feature allows users to keep track of past weight measurements of a client. Since a client is a person, we have set a more-than-reasonable limit of 5000kg for `weight`.

Author: ```suggestion
The weight tracking feature allows users to keep track of past weight measurements of a client. Refer to the list of valid parameters of each input for more details. <REPLACETHIS>
```

We can refer to the sequence diagram [above](#interacting-with-the-note-command) to see how the addition of such fields to clients interact with the components of FitBook.

> For more details on how the `weight` field interact with the `add` and `edit` command, refer [here](#adding-or-editing-a-client).

Reviewer 3: ```suggestion
For more details on how the `weight` field interacts with the `add` and `edit` command, refer [here](#adding-or-editing-a-client).
```

> For more details on how the `weight` field interacts with the `add` and `edit` command, refer [here](#adding-or-editing-a-client).

Reviewer 2: ```suggestion
For more details on how the `weight` field interacts with the `add` and `edit` commands, refer [here](#adding-or-editing-a-client).
```

The activity diagram below illustrates what happens when a user enters a `weight` command.
![WeightCommandActivityDiagram](images/WeightCommandActivityDiagram.png)

#### Height value of a client

> The  `height` feature allows users to track a client's height. Since a client's typically remains constant, we decided not to implement `height` as a trackable value (unlike [weight](#weight-tracking-feature)).

Reviewer: Double spacing between "The" and "`height`"

> The  `height` feature allows users to track a client's height. Since a client's typically remains constant, we decided not to implement `height` as a trackable value (unlike [weight](#weight-tracking-feature)).

Author: ```suggestion
The `height` feature allows users to track a client's height. Since a client's typically remains constant, we decided not to implement `height` as a trackable value (unlike [weight](#weight-tracking-feature)).
```

> The  `height` feature allows users to track a client's height. Since a client's typically remains constant, we decided not to implement `height` as a trackable value (unlike [weight](#weight-tracking-feature)).

Reviewer 3: ```suggestion
The  `height` feature allows users to track a client's height. Since a client's height typically remains constant, we decided not to implement `height` as a trackable value (unlike [weight](#weight-tracking-feature)).
```

> Since a client is a person, we have set a more-than-reasonable limit of 5000cm for `height`.

Reviewer 2: Same comment about valid parameter table.

> Since a client is a person, we have set a more-than-reasonable limit of 5000cm for `height`.

Author: ```suggestion
* Refer to the list of valid parameters of each input for more details. <REPLACETHIS>
```

The `height` field is similar to `note` field, except that the underlying data type is a `Float`, instead of a `String`.

We can refer to the sequence diagram [here](#interacting-with-the-note-command) to see how the addition of such fields to clients interact with the components of FitBook.

> For more details on how the `height` field interact with the `add` and `edit` command, refer [here](#adding-or-editing-a-client).

Reviewer 3: ```suggestion
For more details on how the `height` field interacts with the `add` and `edit` commands, refer [here](#adding-or-editing-a-client).
```
